src/routes.tsx

```diff
@@ -1,26 +1,51 @@
 import React from "react";
+import { useHref } from "react-router-dom";
 
 import { PageLayout } from "./layouts";
 import { EnvironmentDetails } from "./features/environmentDetails";
 import { EnvironmentCreate } from "./features/environmentCreate";
+import { NoEnvironmentSelected } from "./components/NoEnvironmentSelected";
+import { NotFound } from "./components/NotFound";
 
 /**
  * Define URL routes for the single page app
  */
 
+const HardRedirectNotFound = () => {
+  // useHref takes into account React Router basename option. So if basename is
+  // set to "/conda-store", the hook will return "/conda-store/not-found"
+  const url = useHref("/not-found");
+  window.location.replace(url);
```

Comment:
How does this work when using `MemoryRouter`?

Reply:
Good catch. It doesn't. I just pushed a commit that fixes it.